types/relay-test-utils/lib/RelayResolverTestUtils.d.ts

```diff
@@ -0,0 +1,26 @@
+// TODO: move this into relay-runtime
+import { KeyType, KeyTypeData } from 'react-relay/relay-hooks/helpers';
+
+/**
+ * Utility function for testing Relay Resolvers. Pass the resolver function and
+ * the data that will be returned from `readFragment` and it will return the
+ * value that the resolver would derive.
+ *
+ * *Note:* Relay fragment data includes a special `$fragmentType` key which is
+ * impossible for non-Relay code to construct. In tests you can work around
+ * this by passing `null` with a Flow suppression:
+ *
+ * ```
+ * const fragmentData = {
+ *   // Other fields here...
+ *   $fragmentType: (null: any)
+ * };
+ *
+ * const actual = testResolver(resolverFunc, fragmentData);
+ * expect(actual).toEqual(expectedValue)
+ * ```
+ */
+export function testResolver<TKey extends KeyType, Ret>(
+    resolver: (arg: TKey) => Ret,
+    fragmentData: Omit<KeyTypeData<TKey>, ' $fragmentSpreads' | ' $fragmentType'>,
+): Ret;
```

Comment:
I don't understand this function sorry, but the signature looks a bit more complex than: https://github.com/facebook/relay/blob/a93d92231215f680d81482675971b33afd6c1ba0/packages/relay-test-utils/RelayResolverTestUtils.js

I suspect your definition is probably better here, but I don't know.

Reply:
Yeah it's mainly about supplying the data required by a fragment key. As their comment suggest it can be `Key[' $data']` — so we can just do that now. 

The omit is interesting, coz of data supplied by other fragments, not sure howe we unfurl that. But suspect we are fine.